doc/files/package.json.md

```diff
@@ -108,8 +107,34 @@ You can check [the full list of SPDX license IDs](https://spdx.org/licenses/).
 Ideally you should pick one that is
 [OSI](http://opensource.org/licenses/alphabetical) approved.
 
-It's also a good idea to include a LICENSE file at the top level in
-your package.
+If your package is licensed under multiple common licenses, use an SPDX license
+expression syntax version 2.0 string, like this:
+
+    { "license" : "(ISC OR GPL-3.0)" }
+
+If you are using a license that hasn't been assigned an SPDX identifier, or if
+you are using an uncommon or custom license, do not include a "license" string
+in package.json. In those cases especially, but also more generally, it's a good
+idea to include a LICENSE file at the top level of the package.
```

Comment:
Great point. It didn't think about making sure warning scenarios are coextensive with guidelines noncompliance.

I take it `normalize-package-data` should remain a pure function of the `package.json` object, so hitting the file system or the API to `stat` a `LICENSE` is out.

That leaves us at least three ways to go at the problem:

1. SPDX does provide for a way to reference licenses that don't have assigned identifiers, `LicenseRef-X`, but they are designed as references to other RDF objects within a larger package definition, of which license IDs and expressions are just a part. npm could define a special value, like `LicenseRef-LICENSE` that means "check the LICENSE file" in the ecosystem.

2. Acknowledge that failure to indicate license terms is a warning-worthy offense. But ... private modules.

3. Don't issue warnings when `license` is missing. Accept a far greater number of false negatives (should have `license`, doesn't see a warning) for fewer false negatives (shouldn't have `license` for lack of a way to specify it as an SPDX expression, but warn anyway).